app/models/consistency.php

```diff
@@ -0,0 +1,79 @@
+<?php
+namespace Transvision;
+
+// Build arrays for the search form, ignore mozilla.org
+$channel_selector = Utils::getHtmlSelectOptions(
+    array_diff($repos_nice_names, ['mozilla.org']),
+    $repo,
+    true
+);
+
+$target_locales_list = Utils::getHtmlSelectOptions(
+    Project::getRepositoryLocales($repo),
+    $locale
+);
+
+/*
+    Since this is a very heavy view for Aurora (default repo), we display
+    an introductory text when reaching the page and use a paramter to
+    determine if we actually want to display the strings.
+*/
+$display_results = isset($_GET['display']) ? true : false;
+$reference_locale = Project::getReferenceLocale($repo);
+
+if ($display_results) {
+    // Get non-empty strings for reference and locale in this repository
+    $source_strings = array_filter(Utils::getRepoStrings($reference_locale, $repo), 'strlen');
+    $target_strings = array_filter(Utils::getRepoStrings($locale, $repo), 'strlen');
```

Comment:
Thinking about it, array_filter should be done in findDuplicates() and not in the model, that way you get it defined only once and you don't need to think that the data provided has to be filtered for empty strings if the method is used in another view